crates/hyperswitch_domain_models/src/relay.rs

```diff
@@ -0,0 +1,275 @@
+use common_enums::enums;
+use common_utils::{
+    self,
+    errors::{CustomResult, ValidationError},
+    id_type::{self, GenerateId},
+    pii,
+    types::{keymanager, MinorUnit},
+};
+use diesel_models::relay::RelayUpdateInternal;
+use error_stack::ResultExt;
+use masking::{ExposeInterface, Secret};
+use serde::{self, Deserialize, Serialize};
+use time::PrimitiveDateTime;
+
+use crate::{router_data::ErrorResponse, router_response_types};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Relay {
+    pub id: id_type::RelayId,
+    pub connector_resource_id: String,
+    pub connector_id: id_type::MerchantConnectorAccountId,
+    pub profile_id: id_type::ProfileId,
+    pub merchant_id: id_type::MerchantId,
+    pub relay_type: enums::RelayType,
+    pub request_data: Option<RelayData>,
+    pub status: enums::RelayStatus,
+    pub connector_reference_id: Option<String>,
+    pub error_code: Option<String>,
+    pub error_message: Option<String>,
+    #[serde(with = "common_utils::custom_serde::iso8601")]
+    pub created_at: PrimitiveDateTime,
+    #[serde(with = "common_utils::custom_serde::iso8601")]
+    pub modified_at: PrimitiveDateTime,
+    pub response_data: Option<pii::SecretSerdeValue>,
+}
+
+impl Relay {
+    pub fn new(
+        relay_request: &api_models::relay::RelayRequest,
+        merchant_id: &id_type::MerchantId,
+        profile_id: &id_type::ProfileId,
+    ) -> Self {
+        let relay_id = id_type::RelayId::generate();
+        Self {
+            id: relay_id.clone(),
```

Comment:
Why the clone, looks like we aren't using the `relay_id` variable anywhere else?